api-report/tree.api.md

```diff
@@ -4,9 +4,34 @@
 
 ```ts
 
+import { Serializable } from '@fluidframework/datastore-definitions';
+
+// @public
+export interface ITreeCursor {
+    down(key: TreeKey, index: number): TreeNavigationResult;
+    keys: Iterable<TreeKey>;
+    length(key: TreeKey): number;
+    type: TreeType;
+    up(): TreeNavigationResult;
+    value: undefined | Serializable;
+}
+
 // @public (undocumented)
-export const _ = "Force index.d.ts generation from ApiExtractor.";
+export type TreeKey = (number | string) & {
```

Comment:
Is the proposal here that string keys that parsed as integers would be prohibited? Or would one always use the API to read a trait, i.e. one could never just write someTree[key] (since numbers would be coerced to strings if one did that)? Using symbols for identity traits avoids this issue, since those don't get coerced when used as JS property keys.